src/interactive.rs

```diff
@@ -49,7 +49,11 @@ pub fn filter<'a>(
                 .and_then(|val| {
                     val.trim().parse::<usize>()
                         .map_err(|e| {
-                            FilterError::String(format!("unable to parse selection: {}", e))
+                            if val.trim().is_empty() {
```

Comment:
I'd rather chain this earlier and one level higher, e.g. right after `res.unwrap` adding a `.then` which handles empty-string-values.. and then the `.parse` `and_then` won't run because it won't be an Ok value.

Reply:
The `interrupted` error above might as well also be converted over.